ha-jobs-core/src/main/scala/de/kaufhof/hajobs/LockRepository.scala

```diff
@@ -129,17 +132,27 @@ class LockRepository(session: Session, lockTypes: LockTypes) {
 
   /**
    * Returns a list of Locks (jobType + jobId).
+   * To avoid select * statement without key in where clause,
+   * we do multiple select statements with given primary key
    */
   def getAll()(implicit ec: ExecutionContext): Future[Seq[Lock]] = {
-    val query = select().all().from(Table).setConsistencyLevel(LOCAL_QUORUM)
-    session.executeAsync(query).map(rs =>
-      rs.all().foldLeft(Seq.empty[Lock]) { (res, row) =>
-        val lockTypeName = row.getString(LockTypeCol)
-        Try(lockTypes(lockTypeName))
-          .map(lockType => res :+ Lock(lockType, row.getUUID(LockCol)))
-          .getOrElse(res)
-      }
-    )
+    def query(lockType: LockType) = select().all().from(Table)
+      .where(QueryBuilder.eq(LockTypeCol, lockType.name))
+      .setConsistencyLevel(LOCAL_QUORUM)
+
+    val res = lockTypes.getAll.map { lockType =>
+      session.executeAsync(query(lockType)).map(rs =>
+        Option(rs.one).flatMap { row =>
```

Comment:
This changes the semantics - previously if there were 2 locks for a single locktype they'd have been returned, now only *one* is returned - or what happens in the case of 2 locks for the same locktype? (the table structure itself doesn't ensure uniqueness, therefore I'd say it would be more solid not to work with this assumption here)